crates/communication/src/server/outputs/router.rs

```diff
@@ -0,0 +1,367 @@
+                                (cycler_instance.clone(), fields.clone())
+                            })
+                            .collect(),
+                    },
+                )))
+                .await
+                .expect("receiver should always wait for all senders");
+        }
+        OutputRequest::GetNext {
+            id,
+            cycler_instance,
+            ..
+        }
+        | OutputRequest::Subscribe {
+            id,
+            cycler_instance,
+            ..
+        } => {
+            if matches!(request.request, OutputRequest::Subscribe { .. }) {
+                cached_cycler_instances
+                    .insert((request.client.clone(), *id), cycler_instance.clone());
+            }
+
+            match request_channels_of_cyclers.get(cycler_instance) {
+                Some((_fields, request_channel)) => {
+                    request_channel
+                        .send(request)
+                        .await
+                        .expect("receiver should always wait for all senders");
+                }
+                None => {
+                    let error_message = format!("unknown cycler_instance {cycler_instance:?}");
+                    request
+                        .client
+                        .response_sender
+                        .send(Response::Textual(TextualResponse::Outputs(
+                            if matches!(request.request, OutputRequest::GetNext { .. }) {
+                                TextualOutputResponse::GetNext {
+                                    id: *id,
+                                    result: Err(error_message),
+                                }
+                            } else {
+                                TextualOutputResponse::Subscribe {
+                                    id: *id,
+                                    result: Err(error_message),
+                                }
+                            },
+                        )))
+                        .await
+                        .expect("receiver should always wait for all senders");
+                }
+            }
+        }
+        OutputRequest::Unsubscribe {
+            id,
+            subscription_id,
+        } => {
+            let cycler_instance = match cached_cycler_instances
+                .entry((request.client.clone(), *subscription_id))
+            {
+                Entry::Occupied(entry) => entry.remove(),
+                Entry::Vacant(_) => {
+                    request
+                        .client
+                        .response_sender
+                        .send(Response::Textual(TextualResponse::Outputs(
+                            TextualOutputResponse::Subscribe {
+                                id: *id,
+                                result: Err(format!("unknown subscription ID {subscription_id}")),
+                            },
+                        )))
```

Comment:
This should be `TextualOutputResponse::Unsubscribe`

Reply:
Should be fixed.